Disfunction/Assets/Game Scripts/Gun Behaviour/GunMechanics.cs

```diff
@@ -144,13 +148,23 @@ void Shoot() {
             Debug.DrawRay(gunMouth.transform.position, transform.TransformDirection(Vector3.forward) * hit.distance, Color.yellow);
             GameObject obj = Instantiate(debugBall, hit.point, Quaternion.identity);
             obj.transform.SetParent(hit.transform, true);
-            Dummy DummySystem = hit.transform.GetComponent<Dummy>();
-            Debug.Log(DummySystem);
-            if(DummySystem != null)
+            ImpactBearer impactBearer = hit.transform.GetComponent<ImpactBearer>();
+            float impactOver30 = damage - ImpactOver30;
+            float impactOver40 = damage - ImpactOver40;
+            if (impactBearer != null)
             {
-                DummySystem.damage(10);
+                float distanceBetweenGunAndObject = Geometry.GetDistance(gunMouth.transform.position, hit.point);
+
+                if (distanceBetweenGunAndObject < 30)

```

Comment:
Make the hard coded values which is 30 and 40 there as a Global variable with names like meduimRangeOffset/ farRangeOffset